// 7. Reverse Integer

class Solution {
    public int reverse(int x) {

            int rev = 0;
           while (x != 0) {

            //Reverse
            int rem = x % 10;
              x /= 10;

             // Positive overflow
            if (rev > Integer.MAX_VALUE / 10 ||
                (rev == Integer.MAX_VALUE / 10 && rem > 7)) {
                return 0;
            };

             // Negative overflow
            if (rev < Integer.MIN_VALUE / 10 ||
                (rev == Integer.MIN_VALUE / 10 && rem < -8)) {
                return 0;
            }

            rev = rem + 10 * rev;
           };

        return rev;
    }
}